pkg/appgw/health_probes.go

```diff
@@ -168,8 +168,19 @@ func (c *appGwConfigBuilder) getProbeForServiceContainer(service *v1.Service, ba
 		return nil
 	}
 
+	var podContainers []v1.Container
+
+	// let's assume that the pods with the same label can have different health probes across multiple namespaces,
+	// then, get only the containers of a pod from the same namespace as the ingress
+	for k, pod := range podList {
+		if pod.Namespace == backendID.Ingress.Namespace {
+			podContainers = podList[k].Spec.Containers
+			break
+		}
+	}
+
```

Comment:
This is great!  Reading the issue - it makes sense to filter by namespace.
The function within which we are making this change is `getProbeForServiceContainer`
So we generate the health probes for a given service.  Would it make more sense then to look at the name space of the `Service` above instead of the `Ingress`?  I can imagine having an Ingress referencing a services in different namespaces.

So I believe we could achieve the same if we moved this block into the `ListPodsByServiceSelector` function.  We could change the function signature to `c.k8sContext.ListPodsByServiceSelector(service)` (no Selector).  And then we'll have access to both the Selector and the Namespace of the service.

What do you think?

Reply:
Agree, I will make an update. Thank you.

Reply:
@draychev done, Is this you had in mind?

Reply:
This is great!! Thank you!